radix-engine/src/wasm/wasmi.rs

```diff
@@ -295,31 +497,35 @@ impl WasmInstance for WasmiInstance {
         args: Vec<Buffer>,
         runtime: &mut Box<dyn WasmRuntime + 'r>,
     ) -> Result<Vec<u8>, InvokeError<WasmRuntimeError>> {
-        let mut externals = WasmiExternals {
-            instance: self,
-            runtime,
-        };
+        {
+            // set up runtime pointer
+            self.store.data_mut().runtime_ptr = runtime as *mut _ as usize;
```

Comment:
Out of curiosity, how much penalty do we get if we add `Mutex` on the pointer.

Reply:
no penalty observed (the results we observed the other day were false)